### PR TITLE
add GetArchiveState command

### PR DESCRIPTION
**Pull request by DarkPark**

### CODE REVIEW

File: src/obis-observer/commands/uplink/GetArchiveStateResponse.ts [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+/**
+ * IGetArchiveStateResponseParameters command parameters
+ */
+interface IGetArchiveStateResponseParameters extends ICommandParameters {
+    archiveRecordsNumber: number,
+    eldestTime2000: TTime2000,
+    newestTime2000: TTime2000
+}
+
+const COMMAND_ID = 0x29;
```
Comment (DeadByElpy) [inline src/obis-observer/commands/uplink/GetArchiveStateResponse.ts:16]:
```suggestion

const COMMAND_ID = 0x29;
```

File: src/obis-observer/commands/uplink/GetArchiveStateResponse.ts [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+/**
+ * Uplink command.
+ *
+ * @example create command instance from command body hex dump
+ * ```js
+ * import GetArchiveStateResponse from 'jooby-codec/obis-observer/commands/uplink/GetArchiveStateResponse.js';
+ *
+ * const commandBody = new Uint8Array([
+ *     0x10, 0x22, 0x02, 0x2d, 0x19, 0x17, 0xc0, 0x32, 0x41, 0xb2, 0x28, 0xf6, 0x38, 0x42, 0xb2, 0xa8, 0xf6
```
Comment (DeadByElpy) [inline src/obis-observer/commands/uplink/GetArchiveStateResponse.ts:52]: bad example. bytes and parameters not for this command

File: src/obis-observer/commands/uplink/GetArchiveStateResponse.ts [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    static readonly hasParameters = true;
+
+
+    // data - only body (without header)
+    static fromBytes ( data: Uint8Array ) {
+        const buffer = new CommandBinaryBuffer(data);
+
+        return new GetArchiveStateResponse({
+            requestId: buffer.getUint8(),
+            archiveRecordsNumber: buffer.getUint32(false),
```
Comment (DeadByElpy) [inline src/obis-observer/commands/uplink/GetArchiveStateResponse.ts:93]: CommandBinaryBuffer already initiazed in BE bytes format

**Review by DeadByElpy — CHANGES_REQUESTED** [review]

File: src/obis-observer/commands/uplink/GetArchiveStateResponse.ts (hunk 1 above)
Comment (DarkPark) [reply src/obis-observer/commands/uplink/GetArchiveStateResponse.ts:16]: fixed

File: src/obis-observer/commands/uplink/GetArchiveStateResponse.ts (hunk 2 above)
Comment (DarkPark) [reply src/obis-observer/commands/uplink/GetArchiveStateResponse.ts]: fixed

File: src/obis-observer/commands/uplink/GetArchiveStateResponse.ts (hunk 3 above)
Comment (DarkPark) [reply src/obis-observer/commands/uplink/GetArchiveStateResponse.ts]: fixed

**Review by DeadByElpy — APPROVED** [review]

**Pull request merged**